src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpNamingStyleDiagnosticAnalyzer.cs

```diff
@@ -1,16 +1,19 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
 
 namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.NamingStyles
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosticAnalyzerBase
+    internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosticAnalyzerBase<SyntaxKind>
     {
-        public CSharpNamingStyleDiagnosticAnalyzer()
-        {
-
-        }
+        protected override ImmutableArray<SyntaxKind> SupportedSyntaxKinds { get; } =
+            ImmutableArray.Create(
+                SyntaxKind.VariableDeclarator,
+                SyntaxKind.ForEachStatement,
+                SyntaxKind.SingleVariableDesignation,
```

Comment:
I know your PR should have no problems here, but can we have some tests that these new local-variable features do **not** affect members inside tuples.  Thanks!

Reply:
added in 3416dc96ef447e5c624a069d416536a287e6a1a7